### PR TITLE
feat: root navigation tests + extract awala callings to a separate class to make AwalaManager testable + extract Logger to a separate class

### PR DESCRIPTION
**Pull request by mgulyaev10**
I was trying to figure out what causes the issue from the task https://relaycorp.atlassian.net/browse/LTR-116. Have not found it in the code, but just to make sure that everything works as expected and we don't let a user to go to the Home screen before Awala is initialized, I wrote unit tests with different possible cases

So, in this PR:
1. Unit tests for root navigation screen (screen that is placed first in Letro's navigation stack)
2. Callings of Awala SDK's methods extracted to a wrapper class (`AwalaWrapper`) to make `AwalaManager` testable
3. Created `Logger` class to make non-Android classes testable (I refactored only classes that are used in the tests in this PR to make the PR simple, so I didn't change all `Log.method() `callings to the `Logger.method()` callings)

P.S. @gnarea I've checked that library for logging that you shared with me in some of your PRs. I tried to integrate it, but for some reasons, I did not see any logs when I used classes from this library. So I decided to not spend time on figure out the issue, and make a fast solution with my own logger wrapper

### CODE REVIEW

File: app/src/main/java/tech/relaycorp/letro/awala/AwalaManager.kt [hunk 1]
```diff
@@ -132,23 +129,20 @@ class AwalaManagerImpl @Inject constructor(
...
             )
-            Log.i(TAG, "sendMessage() from ${firstPartyEndpoint.nodeId} to ${thirdPartyEndpoint.nodeId}: $outgoingMessage)")
-            GatewayClient.sendMessage(
-                OutgoingMessage.build(
-                    type = outgoingMessage.type.value,
-                    content = outgoingMessage.content,
-                    senderEndpoint = firstPartyEndpoint,
-                    recipientEndpoint = thirdPartyEndpoint,
-                ),
+            logger.i(TAG, "sendMessage() from ${firstPartyEndpoint.nodeId} to ${thirdPartyEndpoint.nodeId}: $outgoingMessage)")
```
Comment (gnarea) [inline app/src/main/java/tech/relaycorp/letro/awala/AwalaManager.kt:141]: I know it's tangential to this PR, but we should stop logging the full `$outgoingMessage` for privacy/security reasons. How about we just log the content type?

File: app/src/main/java/tech/relaycorp/letro/awala/AwalaManager.kt [hunk 2]
```diff
@@ -225,11 +219,11 @@ class AwalaManagerImpl @Inject constructor(
             }
             isReceivingMessages = true
 
-            Log.i(TAG, "start receiving messages...")
-            GatewayClient.receiveMessages().collect { message ->
-                Log.i(TAG, "Receive message: ${message.type}: ($message)")
+            logger.i(TAG, "start receiving messages...")
+            awala.receiveMessages().collect { message ->
+                logger.i(TAG, "Receive message: ${message.type}: ($message)")
```
Comment (gnarea) [inline app/src/main/java/tech/relaycorp/letro/awala/AwalaManager.kt:224]: Same thing here with `$message` -- I think we should skip the contents of the message, but the sender and recipient may be useful

File: app/src/main/java/tech/relaycorp/letro/utils/Logger.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import tech.relaycorp.letro.BuildConfig
+import java.lang.Exception
+import javax.inject.Inject
+
+interface Logger {
+    fun e(tag: String, message: String, exception: Exception)
+    fun w(tag: String, exception: Exception)
+    fun d(tag: String, message: String)
+    fun i(tag: String, message: String)
+    fun println(message: String)
```
Comment (gnarea) [inline app/src/main/java/tech/relaycorp/letro/utils/Logger.kt:13]: How's this different from `Logger.i()`? I can see their implementations are different, but my question is mainly why this is needed.

File: app/src/test/java/tech/relaycorp/letro/main/RootNavigationTest.kt [hunk 4]
```diff
@@ -0,0 +1,307 @@
+package tech.relaycorp.letro.main
+
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.setMain
+import org.junit.jupiter.api.Assertions.assertEquals
```
Comment (gnarea) [inline app/src/test/java/tech/relaycorp/letro/main/RootNavigationTest.kt:8]: Can we please use `kotest` for consistency?

File: app/src/test/java/tech/relaycorp/letro/main/RootNavigationTest.kt [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+            awalaManager = awalaManager,
+            accountRepository = accountRepository,
+            contactsRepository = contactsRepository,
+            conversationsRepository = createConversationsRepository(
+                conversations = listOf(createConversation(conversationId)),
+                messages = listOf(createMessage(conversationId)),
+                awalaManager = awalaManager,
+                contactsRepository = contactsRepository,
+                accountRepository = accountRepository,
+            ),
```
Comment (gnarea) [inline app/src/test/java/tech/relaycorp/letro/main/RootNavigationTest.kt:182]: How about you call `createConversationsRepository()` inside `createViewModel()`? These tests have a lot of repetitive code, and I think this is an easy way to improve this.

However, if you implement my suggestion, you're going to have to add the optional parameters `conversations = emptyList()` and `messages = emptyList()` to `createViewModel()`.

**Review by gnarea — CHANGES_REQUESTED** [review]

**Comment by gnarea** [comment]
Thanks! LGTM overall. Just added some minor points.

File: app/src/main/java/tech/relaycorp/letro/awala/AwalaManager.kt (hunk 1 above)
Comment (mgulyaev10) [reply app/src/main/java/tech/relaycorp/letro/awala/AwalaManager.kt:141]: Sure!

File: app/src/main/java/tech/relaycorp/letro/utils/Logger.kt (hunk 3 above)
Comment (mgulyaev10) [reply app/src/main/java/tech/relaycorp/letro/utils/Logger.kt:13]: I use it only in testing. Doesn't matter, I can use Logger.i() instead

File: app/src/test/java/tech/relaycorp/letro/main/RootNavigationTest.kt (hunk 5 above)
Comment (mgulyaev10) [reply app/src/test/java/tech/relaycorp/letro/main/RootNavigationTest.kt:182]: Sure, nice suggestion!

File: app/src/main/java/tech/relaycorp/letro/awala/AwalaManager.kt (hunk 2 above)
Comment (mgulyaev10) [reply app/src/main/java/tech/relaycorp/letro/awala/AwalaManager.kt:224]: Yes, nice catch! Anyway, this info is useless in logs

**Comment by mgulyaev10** [comment]
> Thanks! LGTM overall. Just added some minor points.

Thanks for reviewing, the code became much prettier! Pushed changes :)

**Review by gnarea — APPROVED** [review]
LGTM! Thanks 🙏🏾

**Pull request merged**